cmake/boilerplate.cmake

```diff
@@ -28,3 +28,6 @@ if(DEFINED NRF_SUPPORTED_BUILD_TYPES)
                 message(FATAL_ERROR "${CMAKE_BUILD_TYPE} variant is not supported")
         endif()
 endif()
+
+add_library(compile_options_final INTERFACE)
```

Comment:
Should we point to the use in `extensions.cmake` somehow to give some rationale?